### ISSUE TITLE
[a11y][Violation][4.1.2 Name, Role, Value] Elements missing valid widget role

### ISSUE THREAD

**Issue by nkn2022** [xtermjs/xterm.js#5325]
**Describe the bug**
Accessibility scanning using `IBM Equal Access Accessibility Checker tool` on Chrome browser revealed some violations in the checkpoint category `4.1.2 Name, Role, Value`. Please find the details of the issue below.

### Issue 1

> **Rule ID**: element_tabbable_role_valid
> **Violation**
> The tabbable element's role 'none' is not a widget role 
>  A tabbable element must have a valid widget role
> 
> **Element location**
> `<div
>     tabindex="0"
>     class="terminal xterm">`
> 
> **What to do**
> If users should not be able to interact with this element using the keyboard, replace the non-widget role, and/or use the explicit tabindex="-1" attribute to remove the element from the keyboard focus (tab) order
> Or, if users should be able to interact with this element using the keyboard, use a valid widget role
> 
> **Screenshot:**

![Image](https://github.com/user-attachments/assets/bbf1c80c-ca2b-4a06-b208-15f824222721)

### Issue 2

> **Rule ID:** input_label_exists
> **Violation**
> Form control element <textarea> has no associated label
> Each form control must have an associated label
> 
> **Element location**
> `<textarea
>     tabindex="0"
>     spellcheck="false"
>     autocapitalize="off"
>     autocorrect="off"
>     class="xterm-helper-textarea">`
> 
> **What to do**
> If the control is a [labelable](https://html.spec.whatwg.org/multipage/forms.html#category-label) element, add a for attribute to a visible label with the id of this control (e.g., <label for="controlID">)
> Or, if the control is a labelable element, put the control inside a visible <label> element
> Or, add an aria-labelledby attribute to the control. It must point to visible text on the page that is meaningful as a label
> Or, if the design cannot have a visible label, provide a label using the aria-label attribute (e.g., aria-label="Activities")
> Or, if the design cannot have a visible label, use the title attribute to provide a label
> 
> **Screenshot:**

![Image](https://github.com/user-attachments/assets/b2582637-fa88-4c33-9f5a-655c60f1fd59)

## Details
- Browser and browser version: Chrome
- OS version: macOS Sonoma 14.7.3
- xterm.js version: 

### Steps to reproduce
1. Install `IBM Equal Access Accessibility Checker` tool on Chrome browser.
2. Open `Terminal` UI (We have a UI that embeds the terminal)
3. Open the dev tools -> Elements -> Accessibility Checker 
4. Run scan by clicking the `Scan` button
5. See 2 violations reported after the scan

Kindly help take a look.

**Comment by nkn2022** [comment]
For the first issue the element is created by the following code:
https://github.com/xtermjs/xterm.js/blob/a260f7d2889142d6566a66cb9856a07050dea611/src/browser/CoreBrowserTerminal.ts#L417-L420

For the 2nd issue the element is created by the following code https://github.com/xtermjs/xterm.js/blob/a260f7d2889142d6566a66cb9856a07050dea611/src/browser/CoreBrowserTerminal.ts#L440-L441

**Comment by Tyriar** [comment]
> For the 2nd issue the element is created by the following code

@nkn2022 the line after you links sets it, the aria-label is defined by embedders by setting `Terminal.strings.promptLabel`. Here's how VS Code does it for example:

https://github.com/microsoft/vscode/blob/bf35dd67fa0b4f5dd4cda28c2a14681ce422a5ce/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L770-L771

**Comment by Tyriar** [comment]
> For the first issue the element is created by the following code:

I think the latest version of the code doesn't do this.

**Comment by nkn2022** [comment]
Thanks, @Tyriar, for the inputs. We will try your recommendations.
